testkit/javadsl/src/main/scala/com/lightbend/lagom/javadsl/testkit/ServiceTest.scala

```diff
@@ -348,7 +424,15 @@ object ServiceTest {
       awaitPersistenceInit(system)
     }
 
-    new TestServer(assignedPort, application, srv)
+    // TODO: review this when SSLContext provider is promoted to play or ssl-config
+    val sslContext =
+      if (setup.ssl) {
+        Some(new LagomDevModeSSLEngineProvider(application.environment().rootPath).sslContext)
+      }else{
+        None
+      }
+
+    new TestServer(assignedPort, application, srv, sslPort, sslContext)
```

Comment:
There's quite some duplication here from the scala counterpart. This may be promoted to `testkit-core`.

Reply:
Should we also promote `Setup` and `SetupImpl` or is it out of scope for this PR?

Reply:
Hmmm, well. I'm reconsidering my previous comment. 
These cases (TestServer, Setup, ...)  are public API and the approach in Lagom is to write twice. We could try, I guess.